File: mobile/src/main/java/org/horizontal/tella/mobile/data/repository/FeedbackRepository.java
package org.horizontal.tella.mobile.data.repository;

import androidx.annotation.NonNull;

import io.reactivex.Completable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import org.horizontal.tella.mobile.data.entity.mapper.EntityMapper;
import org.horizontal.tella.mobile.data.rest.FeedbackApi;
import org.horizontal.tella.mobile.domain.entity.Feedback;
import org.horizontal.tella.mobile.domain.repository.IFeedbackRepository;


public class FeedbackRepository implements IFeedbackRepository {

    @Override
    public Completable sendFeedback(@NonNull Feedback feedback) {
        return FeedbackApi.getApi().sendFeedback(new EntityMapper().transform(feedback))
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .onErrorResumeNext(throwable -> Completable.error(new ErrorBundle(throwable)));
    }
}

File: mobile/src/main/java/rs/readahead/washington/mobile/domain/repository/feedback/FeedbackRepository.kt
package rs.readahead.washington.mobile.domain.repository.feedback

import io.reactivex.Single
import rs.readahead.washington.mobile.data.entity.feedback.FeedbackBodyEntity
import rs.readahead.washington.mobile.domain.entity.feedback.FeedbackPostResult


interface FeedbackRepository {
    fun submitFeedback(
            feedbackBody: FeedbackBodyEntity
    ): Single<FeedbackPostResult>
}